core/block.js

```diff
@@ -1445,3 +1445,19 @@ Blockly.Block.prototype.allInputsFilled = function(opt_shadowBlocksAreFilled) {
 
   return true;
 };
+
+/**
+ * @return {string} A description of this block using type name and id.
+ */
+Blockly.Block.prototype.describe = function() {
```

Comment:
.toString() is already used:

> var b = Blockly.mainWorkspace.newBlock('controls_if')
> b
B…y.BlockSvg {svgGroup_: g, svgPathDark_: path, svgPath_: path, svgPathLight_: path, rendered: false…}
> b.toString()
> "if ? do ?"